File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
class Solution {
    public int solution(String s) {
        String[] numstr = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

        for(int i=0;i<numstr.length;i++) {
        	if(s.contains(numstr[i])) {
        		s = s.replace(numstr[i], Integer.toString(i));
        	}
        }
        return Integer.parseInt(s);
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
// - Object.keys()
// - Array.prototype.forEach()
// - String.prototype.replaceAll()
function solution(s) {
    const dict = {
        zero: 0,
        one: 1,
        two: 2,
        three: 3,
        four: 4,
        five: 5,
        six: 6,
        seven: 7,
        eight: 8,
        nine: 9
    };
    
    Object.keys(dict).forEach(item => s = s.replaceAll(item, dict[item])); 
    // ["zero", "one", ...]
    return +s;
}